yb-voyager/src/tgtdb/yugabytedb_test.go

```diff
@@ -290,3 +291,189 @@ func TestGetPrimaryKeyConstraintNames(t *testing.T) {
 		testutils.AssertEqualStringSlices(t, tt.expectedPKNames, pkNames)
 	}
 }
+
+func TestCollectPgStatStatements_BasicSelectQueries(t *testing.T) {
+	// Test data: queries and their expected execution counts across nodes
+	testQueries := map[string]struct {
+		text          string
+		parameterized string
+		execCounts    []int // executions per node [node0, node1, node2]
+	}{
+		"simple_avg": {
+			text:          `SELECT AVG(salary) FROM test_pgss.employees`,
+			parameterized: `SELECT AVG(salary) FROM test_pgss.employees`,
+			execCounts:    []int{1, 0, 0}, // only on node 0
+		},
+		"simple_count": {
+			text:          `SELECT COUNT(*) FROM test_pgss.employees`,
+			parameterized: `SELECT COUNT(*) FROM test_pgss.employees`,
+			execCounts:    []int{0, 1, 0}, // only on node 1
+		},
+		"repeated_query": {
+			text:          `SELECT 'test_merge' as marker, COUNT(*) FROM test_pgss.employees`,
+			parameterized: `SELECT $1 as marker, COUNT(*) FROM test_pgss.employees`,
+			execCounts:    []int{2, 3, 1}, // distributed across all nodes
+		},
+	}
+
+	// Setup test environment
+	testYugabyteDBTargetCluster.ExecuteSqls(
+		`CREATE SCHEMA IF NOT EXISTS test_pgss;`,
+		`CREATE TABLE IF NOT EXISTS test_pgss.employees (
+			id INT PRIMARY KEY, name TEXT, salary INT
+		);`,
+		`INSERT INTO test_pgss.employees VALUES
+			(1, 'Alice', 75000), (2, 'Bob', 55000), (3, 'Charlie', 60000);`,
+		`CREATE EXTENSION IF NOT EXISTS pg_stat_statements;`,
+		`SELECT pg_stat_statements_reset();`,
+	)
+	defer testYugabyteDBTargetCluster.ExecuteSqls(`DROP SCHEMA test_pgss CASCADE;`)
+
+	// Execute queries on different nodes
+	for queryName, query := range testQueries {
+		for nodeIdx, execCount := range query.execCounts {
+			if execCount == 0 {
+				continue
+			}
+
+			conn, err := testYugabyteDBTargetCluster.GetNodeConnection(nodeIdx)
+			require.NoError(t, err, "Failed to connect to node %d for %s", nodeIdx, queryName)
+			defer conn.Close()
```

Comment:
We can probably just close this connection after th for loop executing the queries right? We wont need these same node connections again in the function right?
defer will make it close at the end of the function itself. Probably better to close it right after usage completes?